packages/medusa/src/models/order.js

```diff
@@ -1,26 +1,28 @@
-/*******************************************************************************
- *
- ******************************************************************************/
 import mongoose from "mongoose"
 import { BaseModel } from "medusa-interfaces"
 
 import LineItemSchema from "./schemas/line-item"
 import PaymentMethodSchema from "./schemas/payment-method"
 import ShippingMethodSchema from "./schemas/shipping-method"
 import AddressSchema from "./schemas/address"
+import DiscountModel from "./discount"
 
 class OrderModel extends BaseModel {
   static modelName = "Order"
 
   static schema = {
-    canceled: { type: Boolean, default: false },
-    archived: { type: Boolean, default: false },
+    // pending, completed, archived, cancelled
```

Comment:
When is an order `completed`? I don't see it anywhere else? 

I think it is a nice status to have so maybe we should automatically mark an order as `completed` when fulfillment has been created and payment has been captured. So in each of `capturePayment` and `createFulfillment` we check if the status is different from the defaults (`awaiting`, `not_fulfilled`) if so we also set `status` to `completed`.

Reply:
Also a good point. Well spotted!